### PR TITLE
Potential solution for getting field values from request/response pojos

### PR DESCRIPTION
**Pull request by spfink**
Basically code generating reflection instead of using reflection.

### CODE REVIEW

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java [hunk 1]
```diff
@@ -579,6 +580,57 @@ public String toString() {
         return sb.toString();
     }
 
+    public <T> Optional<T> getValueForField(String fieldName, Class<T> clazz) {
+        switch (fieldName.toLowerCase()) {
+        case "stringMember":
```
Comment (dagnir) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java]: The switch is on the lowercased fieldname so these need to be lowercased too

**Comment by dagnir** [comment]
I'm okay with this, nice and simple. The intended contract for `fieldName` is that it's exactly the same as it appears in the service model right? If so I think we need some extra logic for that when constructing the switch statement for cases where we rename members because they conflict with keywords for example

**Comment by shorea** [comment]
+1 to using member name rather then field name. Member name is a strong and consistent contract. Field names are internal and are subject to change at any time.

**Comment by spfink** [comment]
Updated to use member names. Also updated EndpointAddressInterceptor.java to use this new method.

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java [hunk 2]
```diff
@@ -578,6 +579,57 @@ public String toString() {
         return sb.toString();
     }
 
+    public <T> Optional<T> getValueForField(String fieldName, Class<T> clazz) {
```
Comment (shorea) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java]: I'm a little wary of the lowercasing of the member name. I'd personally rather not do it until there's a need and for S3 it looks like there's not.

Comment (spfink) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java]: I'm fine either way. Only did it so people wouldn't have to worry about using pascal case but probably not a big issue.

Comment (spfink) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java]: Updated.

**Pull request merged**